wordpress-mobile/WordPress-FluxC-Android
# Notifications: Remove localSiteId and replace with remoteSiteId

Fixes #1293 by removing `localSiteId` from the various notification-related models and replacing it with `remoteSiteId` where applicable. All tests have also been updated. 

**Note:** PR #1292 must be approved and merged before this one can be reviewed.

## To Test
- Please make sure to test upgrading from the version in develop (be sure to pull notifications) and then upgrade to this version and pull notifications. 
- Run all the automated tests
- Run all the notifications tests in the example app
